File: CareerDayApi/Entities/Session.cs
namespace CareerDayApi.Entities
{
    public class Session
    {
        public int Id { get; set; }
        public Classroom Classroom { get; set; }
        public List<Student> Students { get; set; } = [];
        public List<Speaker> Speakers { get; set; } = [];
        public int Period { get; set; }
        public Career Subject { get; set; }

        public void AddStudent(Student student)
        {
            if (Students.All(newStudent => newStudent.Id != student.Id)) {
                Students.Add(student);
            }
        }

        public void RemoveStudent(int studentId)
        {
            var student = Students.FirstOrDefault(student => student.Id == studentId);
            if (student == null) return;
            Students.Remove(student);
        }
    }
}

File: Client/src/app/models/session.ts
import { Career } from "./career"
import { Classroom } from "./classroom"
import { Speaker } from "./speaker"
import { Student } from "./student"

export interface Session {
    id: number
    classroom?: Classroom
    students: Student[]
    speakers: Speaker[]
    period: number
    subject: Career
    eventId: number
}